### PR TITLE
Fix no "Available versions" in restore dialog for Filenames with spaces

### PR DESCRIPTION
**Pull request by MSP-Stev**
https://bugs.bareos.org/view.php?id=1265

### CODE REVIEW

**Comment by arogge** [comment]
Would you mind testing this with a file that has a double-quotes in the name? It looks like that might be broken, too.

File: webui/module/Restore/src/Restore/Model/RestoreModel.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ public function getFiles(&$bsock=null, $jobid=null, $pathid=null) {
     */
    public function getFileVersions(&$bsock=null, $clientname=null, $pathid=null, $filename=null) {
       if(isset($bsock)) {
-         $cmd = '.bvfs_versions jobid=0 client='.$clientname.' pathid='.$pathid.' fname='.$filename;
+         $cmd = '.bvfs_versions jobid=0 client='.$clientname.' pathid='.$pathid.' fname="'.$filename.'"';
```
Comment (fbergkemper) [inline webui/module/Restore/src/Restore/Model/RestoreModel.php:158]:
```suggestion
         $cmd = '.bvfs_versions jobid=0 client='.$clientname.' pathid='.$pathid.' fname="'.addslashes($filename).'"';
```
We should add backslashes before characters that need to be escaped in the fname string, too.

**Review by fbergkemper — CHANGES_REQUESTED** [review]
Thanks for the contribution. Can you please apply the suggested change?

**Comment by fbergkemper** [comment]
Suggestion applied by an additional commit.

**Comment by fbergkemper** [comment]
MSP-Stev, if you would like to get mentioned in the Authors file with your real name, please let us know by a comment here.

**Comment by MSP-Stev** [comment]
Sorry for the late response. i will test your suggestion on our system. my name is Stev Dubau :)

**Comment by fbergkemper** [comment]
No problem, Stev. Going to add you to the Authors file, thanks again for the contribution.

**Comment by MSP-Stev** [comment]
you are welcome.

**Pull request merged**